js/main.js

```diff
@@ -404,39 +404,56 @@ function initiate() {
             );
         }
 
+        function addAllTypicalWarningsInFragment(fragment, fragmentIndex, badness, errorCode, type) {
+            var badPos = 0;
+            var offset = 0;
+            var subFragment = fragment.substring(0);
+            while (true) {
```

Comment:
So the previous version of this code would test the regexp only once while you apply it sequentially? Why don't you do the same trick with `exec` and  `while result != null` as below?